--- python/build_targets.bzl
+++ python/build_targets.bzl
@@ -522 +522,10 @@
-    )
+    )
+    proto_lang_toolchain(
+        name = "pyi_toolchain",
+        command_line = "--pyi_out=%s",
+        progress_message = "Generating Pyi proto_library %{label}",
+        runtime = ":protobuf_python",
+        # NOTE: This isn't *actually* public. It's an implicit dependency of py_proto_library,
+        # so must be public so user usages of the rule can reference it.
+        visibility = ["//visibility:public"],
+    )